packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -533,11 +533,16 @@ abstract class FlutterCommand extends Command<void> {
 
     devices = devices.where((Device device) => device.isSupported()).toList();
     // If the user has not specified all devices and has multiple connected
-    // then filter then list by those supported in the current project. If
-    // this ends up with a single device we can proceed as normal.
+    // then filter then list by those supported in the current project and
+    // remove non-ephemeral device types. If this ends up with a single
+    // device we can proceed as normal.
     if (devices.length > 1 && !deviceManager.hasSpecifiedAllDevices && !deviceManager.hasSpecifiedDeviceId) {
       final FlutterProject flutterProject = FlutterProject.current();
       devices.removeWhere((Device device) => !device.isSupportedForProject(flutterProject));
+
+      if (devices.any((Device device) => device.ephemeral)) {
```

Comment:
nit: why not just
```dart
devices = devices.where((Device device) => !device.ephemeral).toList();
```
to be consistent with the code above?

Reply:
If we only have two non-ephemeral devices connected we don't want to remove both.

Reply:
Though I could reassign the list instead of removing, I don't think it makes a big difference here